### PR TITLE
Fixes #543 - remove cursor from URLBar

### LINKED ISSUE #543: Remove cursor from URLBars
Since you can't select / copy / paste and you get the cursor in the keyboard view. Prevent the cursor and character navigation on the URLbars

### PR DESCRIPTION
**Pull request by boek**
image

Now the URLBar gets and relieves focus like any other selectable element in the overlay

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/widget/BrowserNavigationOverlay.kt [hunk 1]
```diff
@@ -86,6 +86,11 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
             if (it == null) { return@forEach }
             it.setTint(ContextCompat.getColor(context, R.color.button_default))
         }
+
+        navUrlInput.compoundDrawablesRelative.forEach {
+            if (it == null) { return@forEach }
+            it.setTint(ContextCompat.getColor(context, R.color.button_text_color))
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/widget/BrowserNavigationOverlay.kt]: nit: `it?.setTint(...` and rm null check, here and above.

fwiw, I'm not sure if return will return from the current function call (likely) or the entire `forEach` iteration.

**Comment by lime124** [comment]
that is, uh, very teal @boek :)

i'm still leaning towards using a heavy stroke for the highlight color over the enter area - which is a lot of blue. i mocked it up super fast using some of the colors we talked about today for the buttons

screen shot 2018-02-28 at 9 32 46 pm

to be fair, i haven't loaded it up on the tv yet.

**Comment by boek** [comment]
@lime124 

screen

**Comment by boek** [comment]
With the stroke instead:

screen

**Comment by boek** [comment]
The active/focus color is probably too similar still?

**Comment by boek** [comment]
Put that border that @shorlander had. Definitely a nice touch 👌 
screen

**Comment by boek** [comment]
Sorry @mcomella, this PR definitely scope creeped into a handful of UX review items :( If you want me to break it up I can

File: app/src/main/java/org/mozilla/focus/widget/InlineAutocompleteEditText.java [hunk 2]
```diff
@@ -658,4 +641,19 @@ private void setIsKeyFromRemoteAppOrSoftKeyboardClear(final boolean isKeyFromRem
             removeAutocomplete(getText()); // Perhaps not strictly necessary.
         }
     }
+
+
```
[the remark this answers is not in the document]
Comment (boek) [reply app/src/main/java/org/mozilla/focus/widget/InlineAutocompleteEditText.java]: I was running ktlint, didn't know there was another one!

**Comment by boek** [comment]
Changed it to a blacklist 👍

File: app/src/main/java/org/mozilla/focus/widget/InlineAutocompleteEditText.java [hunk 3]
```diff
@@ -658,4 +641,24 @@ private void setIsKeyFromRemoteAppOrSoftKeyboardClear(final boolean isKeyFromRem
             removeAutocomplete(getText()); // Perhaps not strictly necessary.
         }
     }
+
+    @Override
+    public boolean dispatchKeyEvent(KeyEvent event) {
+        // This prevents the selector from entering the text area
+        final int keyCode = event.getKeyCode();
+        final boolean shouldDispatch = keyCode != KeyEvent.KEYCODE_DPAD_DOWN
+                || keyCode != KeyEvent.KEYCODE_DPAD_DOWN_LEFT
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/widget/InlineAutocompleteEditText.java]: nit: for future reference: || should be at the end of the line, not the beginning.

**Pull request merged**